cypress/e2e/awx/resources/workflowVisualizer.cy.ts

```diff
@@ -166,54 +141,274 @@ describe('Workflow Job templates visualizer', () => {
+        cy.selectDropdownOptionByResourceName('project', `${project.name}`);
+        cy.selectDropdownOptionByResourceName('node-convergence', 'All');
+        cy.get('[data-cy="node-alias"]').type('Project Node');
+        cy.clickButton('Next');
+        cy.clickButton('Finish');
+        cy.clickButton('Save');
+        cy.get('[data-cy="alert-toaster"]').should(
+          'have.text',
+          'Success alert:Successfully saved workflow visualizer'
+        );
+
+        cy.get('button[data-cy="workflow-visualizer-toolbar-close"]').click();
+
+        cy.get('[data-cy="page-title"]').should('have.text', `${workflowJobTemplate.name}`);
+      });
+  });
+  it('Can delete one single node and save the visualizer', function () {
+    let projectNode: WorkflowNode;
+    cy.createAwxWorkflowVisualizerProjectNode(workflowJobTemplate, project).then((projNode) => {
+      projectNode = projNode;
+      cy.createAwxWorkflowVisualizerApprovalNode(workflowJobTemplate).then(() => {
+        cy.visit(`/templates/workflow_job_template/${workflowJobTemplate?.id}/visualizer`);
+        cy.contains('Workflow Visualizer').should('be.visible');
+        cy.get(`g[data-id=${projectNode.id}] .pf-topology__node__action-icon`).click({
+          force: true,
+        });
+        cy.get('li[data-cy="add-node-and-link"]').click();
+
+        cy.selectDropdownOptionByResourceName('node-type', 'Job Template');
+        cy.selectDropdownOptionByResourceName('job-template-select', `${jobTemplate.name}`);
+        cy.selectDropdownOptionByResourceName('node-status-type', 'Always');
+        cy.selectDropdownOptionByResourceName('node-convergence', 'All');
+        cy.get('[data-cy="node-alias"]').type('Test Node');
+        cy.clickButton('Next');
+        cy.clickButton('Finish');
+        cy.get('g[data-id="3-unsavedNode"]').should('have.text', 'Test Node');
+        cy.get(`g[data-id=${projectNode.id}-3-unsavedNode]`).should('have.text', 'Run always');
+        cy.get(`g[data-id=${projectNode.id}] .pf-topology__node__action-icon`).click({
+          force: true,
+        });
+        cy.get('li[data-cy="remove-node"]').click();
+        cy.clickModalConfirmCheckbox();
+        cy.clickModalButton('Remove');
+        cy.clickModalButton('Close');
+        cy.clickButton('Save');
+        cy.get('[data-cy="alert-toaster"]').should(
+          'have.text',
+          'Success alert:Successfully saved workflow visualizer'
+        );
+        cy.get('button[data-cy="workflow-visualizer-toolbar-close"]').click();
+        cy.get('[data-cy="page-title"]').should('have.text', `${workflowJobTemplate.name}`);
+      });
+    });
+  });
+  it.skip('Can remove all existing nodes on a visualizer using the button in the toolbar kebab, save the visualizer, then add 2 new nodes and save the visualizer again', function () {
+    cy.createAwxWorkflowVisualizerProjectNode(workflowJobTemplate, project).then((projectNode) => {
```

Comment:
Just wondering why this test is skipped?

Reply:
It fails currently, but should pass before we merge into main